devbin/generate_gcp_ar_cleanup_policy.py

```diff
@@ -0,0 +1,140 @@
+        if self.tag_prefixes is not None:
+            condition['tagPrefixes'] = self.tag_prefixes
+        if self.version_name_prefixes is not None:
+            condition['versionNamePrefixes'] = self.version_name_prefixes
+        if self.package_name_prefixes is not None:
+            condition['packageNamePrefixes'] = self.package_name_prefixes
+        if self.older_than:
+            condition['olderThan'] = self.older_than
+        if self.newer_than:
+            condition['newerThan'] = self.newer_than
+        return data
+
+
+class ConditionalKeepPolicy(CleanupPolicy):
+    def __init__(self,
+                 name: str,
+                 tag_state: str,
+                 *,
+                 tag_prefixes: Optional[List[str]] = None,
+                 version_name_prefixes: Optional[List[str]] = None,
+                 package_name_prefixes: Optional[List[str]] = None,
+                 older_than: Optional[str] = None,
+                 newer_than: Optional[str] = None):
+        self.name = name
+        self.tag_state = tag_state
+        self.tag_prefixes = tag_prefixes
+        self.version_name_prefixes = version_name_prefixes
+        self.package_name_prefixes = package_name_prefixes
+        self.older_than = older_than
+        self.newer_than = newer_than
+
+    def to_dict(self):
+        data = {'name': self.name, 'action': {'type': 'Keep'}, 'condition': {'tagState': self.tag_state}}
+        condition = data['condition']
+        if self.tag_prefixes is not None:
+            condition['tagPrefixes'] = self.tag_prefixes
+        if self.version_name_prefixes is not None:
+            condition['versionNamePrefixes'] = self.version_name_prefixes
+        if self.package_name_prefixes is not None:
+            condition['packageNamePrefixes'] = self.package_name_prefixes
+        if self.older_than:
+            condition['olderThan'] = self.older_than
+        if self.newer_than:
+            condition['newerThan'] = self.newer_than
+        return data
+
+
+class MostRecentVersionKeepPolicy(CleanupPolicy):
+    def __init__(self,
+                 name: str,
+                 package_name_prefixes: List[str],
+                 keep_count: int):
+        self.name = name
+        self.package_name_prefixes = package_name_prefixes
+        self.keep_count = keep_count
+
+    def to_dict(self):
+        data = {
+            'name': self.name,
+            'action': {'type': 'Keep'},
+            'mostRecentVersions': {
+                'packageNamePrefixes': self.package_name_prefixes,
+                'keepCount': self.keep_count
+            }
+        }
+        return data
+
+
+third_party_images_fp = 'docker/third-party/images.txt'
+third_party_packages = []
+with open(third_party_images_fp, 'r') as f:
+    for image in f:
+        image = image.strip()
+        package = image.split(':')[0]
+        if package not in third_party_packages:
+            third_party_packages.append(package)
```

Comment:
Can we keep only the tags listed there?

Reply:
I guess that means we need #-of-package policies, which is currently 11.

Reply:
I grabbed all of the tags if they exist, but I'm concerned because these are tag prefix matches so it's not 100% specific to the package.

Reply:
Yeah, I think if we can't do something exact we should do nothing at all. Better to not have a false sense of security.